src/pydicom/dataset.py

```diff
@@ -2230,20 +2196,12 @@ def __setattr__(self, name: str, value: Any) -> None:
             # XXX note if user misspells a dicom data_element - no error!!!
             object.__setattr__(self, name, value)
 
-    def _set_file_meta(self, value: Optional["Dataset"]) -> None:
-        if value is not None and not isinstance(value, FileMetaDataset):
-            if config._use_future:
-                raise TypeError(
-                    "Pydicom Future: Dataset.file_meta must be an instance "
-                    "of FileMetaDataset"
-                )
+    def _set_file_meta(self, value: Optional["FileMetaDataset"]) -> None:
+        if value is not None and not isinstance(value, Dataset):
+            raise TypeError("'Dataset.file_meta' must be a 'FileMetaDataset' instance")
 
-            FileMetaDataset.validate(value)
-            warnings.warn(
-                "Starting in pydicom 3.0, Dataset.file_meta must be a "
-                "FileMetaDataset class instance",
-                DeprecationWarning,
-            )
+        if isinstance(value, Dataset):
+            value = FileMetaDataset(value)
```

Comment:
Sorry, it should be:
```python
        if value is None:
            self.__dict__["file_meta"] = None
            return

        if not isinstance(value, Dataset):
            raise TypeError("'Dataset.file_meta' must be a 'FileMetaDataset' instance")

        if not isinstance(value, FileMetaDataset):
            # Also validates to make sure only group 2 elements
            value = FileMetaDataset(value)

        self.__dict__["file_meta"] = value
```
I wanted to allow setting file_meta using `Dataset` with only group 2, which would be converted to FileMetaDataset, since it seemed less breaking.

Reply:
Right, this is exactly the same semantics as the snippet I wrote, but better readable. Yes, please do it that way.

Reply:
Done

Reply:
I always like 'less breaking' so great idea and it reads very well now.